plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -43,8 +45,46 @@ import { PACKAGE_NAME, PACKAGE_VERSION } from './version';
 import { SpanNames } from './enums/SpanNames';
 
 export class PgInstrumentation extends InstrumentationBase<PgInstrumentationConfig> {
+  private _connectionsCount!: UpDownCounter;
+  private _connectionPendingRequests!: UpDownCounter;
+  // Pool events connect, acquire, release and remove can be called
+  // multiple times without changing the values of total, idle and waiting
+  // connections. The _connectionsCounter is used to keep track of latest
+  // values and only update the metrics _connectionsCount and _connectionPendingRequests
+  // when the value change.
+  private _connectionsCounter: utils.poolConnectionsCounter = {
+    used: 0,
+    idle: 0,
+    pending: 0,
+  };
+
   constructor(config: PgInstrumentationConfig = {}) {
     super(PACKAGE_NAME, PACKAGE_VERSION, config);
+    this._setMetricInstruments();
+  }
+
+  override setMeterProvider(meterProvider: MeterProvider) {
+    super.setMeterProvider(meterProvider);
+    this._setMetricInstruments();
+  }
+
+  private _setMetricInstruments() {
+    this._connectionsCount = this.meter.createUpDownCounter(
+      'db.client.connection.count',
+      {
+        description:
+          'The number of connections that are currently in state described by the state attribute.',
+        unit: '{connection}',
+      }
+    );
```

Comment:
Took me quite a while to figure that out, the test is failing as we don't reset the internally kept state when the MeterProvider is reset. This means the internal state assumes the counter is at `1` when it really is at `0` as the old counter which was at `1` has been replaced by a new one.

```suggestion
    this._connectionsCounter = {
      idle: 0,
      pending: 0,
      used: 0,
    }
    this._connectionsCount = this.meter.createUpDownCounter(
      'db.client.connection.count',
      {
        description:
          'The number of connections that are currently in state described by the state attribute.',
        unit: '{connection}',
      }
    );
```

Another reason why I think the semconv should switch to a synchronous gauge instead for this. :sweat_smile: